addOns/ascanrulesAlpha/src/test/java/org/zaproxy/zap/extension/ascanrulesAlpha/SuspiciousInputTransformationScriptTest.java

```diff
@@ -0,0 +1,400 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascanrulesAlpha;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+
+import fi.iki.elonen.NanoHTTPD;
+import fi.iki.elonen.NanoHTTPD.IHTTPSession;
+import fi.iki.elonen.NanoHTTPD.Response;
+import java.nio.file.Path;
+import java.util.Optional;
+import org.junit.jupiter.api.Test;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.addon.commonlib.CommonAlertTag;
+import org.zaproxy.zap.control.AddOn;
+import org.zaproxy.zap.extension.graaljs.GraalJsActiveScriptScanRuleTestUtils;
+import org.zaproxy.zap.testutils.NanoServerHandler;
+
+class SuspiciousInputTransformationScriptTest extends GraalJsActiveScriptScanRuleTestUtils {
+    @Override
+    public Path getScriptPath() throws Exception {
+        return Path.of(
+                getClass()
+                        .getResource("/scripts/scripts/active/SuspiciousInputTransformation.js")
+                        .toURI());
+    }
+
+    @Test
+    void shouldReturnExpectedMappings() {
+        assertThat(rule.getId(), is(equalTo(100044)));
+        assertThat(rule.getName(), is(equalTo("Suspicious Input Transformation")));
+        assertThat(rule.getCategory(), is(equalTo(Category.INJECTION)));
+        assertThat(rule.getRisk(), is(equalTo(Alert.RISK_HIGH)));
+        assertThat(rule.getCweId(), is(equalTo(20)));
+        assertThat(rule.getWascId(), is(equalTo(20)));
+        assertThat(
+                rule.getAlertTags().keySet(),
+                containsInAnyOrder(
+                        CommonAlertTag.OWASP_2021_A03_INJECTION.getTag(),
+                        CommonAlertTag.OWASP_2017_A01_INJECTION.getTag()));
+        assertThat(rule.getStatus(), is(equalTo(AddOn.Status.alpha)));
+    }
+
+    @Test
+    void shouldRaiseAlertOnQuoteConsumption() throws Exception {
+        // Given
+        String path = "/quoteConsumption";
+        nano.addHandler(new TransformationHandler(path));
+        HttpMessage msg = getHttpMessage(path + "?param=value");
+
+        // When
+        rule.init(msg, parent);
```

Comment:
Kinda nit picky but I think most of the Java rule tests have the init as part of the // Given block

Reply:
Yeah I was thinking that too, will tweak